File: ride.pas
{
ID: wind1740
PROG: ride 
LANG: PASCAL
}
var str1,str2:string;
	
function num(strr:string):longint;
var i:longint;
begin
  num:=1;
  for i:=1 to length(strr) do 
  begin
    num:=(num*((ord (strr[i])-ord('A')+1) mod 47)) mod 47;
  end;
end;

begin
  assign (input,'ride.in');
  assign (output,'ride.out');
  reset (input);
  rewrite (output);
  readln (str1);
  readln (str2);
  if num(str1)=num(str2) then
    writeln ('GO')
  else
    writeln ('STAY');  
  
  close (input);
  close (output);
  
end.

File: ride.java
/*
ID: wind1740
LANG: JAVA
PROG: ride 
*/

import java.io.*;
import java.util.*;



public class ride
{
	public static int num(String strr)
	{
		int j=1;
		for (int i=0;i<strr.length();i++)
		{
			
			j=j*(((int)strr.charAt(i) -(int)'A'+1)%47)%47;
		}
		return (j);
	}
	public static void main(String[] args) throws FileNotFoundException
    {
		Scanner sc = new Scanner(new File ("ride.in"));
		Formatter out = new Formatter(new File("ride.out"));
		
		String str1=sc.nextLine();
		String str2=sc.nextLine();
		
		
		
		if (num(str1)==num(str2))
		{
			out.format("GO%n");
		}
		else
		{
			out.format("STAY%n");
		}
		sc.close();
		out.close();
	}
}